# Import libraries and data

In [1]:
import pandas as pd 
from mlxtend.frequent_patterns import apriori, association_rules 

In [2]:
POIdata_cityA = pd.read_csv('POIdata_cityA.csv')
POIdata_cityB = pd.read_csv('POIdata_cityB.csv')
POIdata_cityC = pd.read_csv('POIdata_cityC.csv')
POIdata_cityD = pd.read_csv('POIdata_cityD.csv')

POI_datacategories = pd.read_csv('POI_datacategories.csv', header=None)

In [3]:
POIdata_cityA.head()

x  y  category  POI_count
0  1  1        74          4
1  1  1        48          4
2  1  1        79          2
3  1  1        69          2
4  1  1        73          1

In [4]:
POI_datacategories.head()

0
0                 Food
1             Shopping
2        Entertainment
3  Japanese restaurant
4   Western restaurant

Check for any na values

In [5]:
POIdata_cityA.isna().sum()

x            0
y            0
category     0
POI_count    0
dtype: int64

In [6]:
POIdata_cityB.isna().sum()

x            0
y            0
category     0
POI_count    0
dtype: int64

In [7]:
POIdata_cityC.isna().sum()

x            0
y            0
category     0
POI_count    0
dtype: int64

In [8]:
POIdata_cityD.isna().sum()

x            0
y            0
category     0
POI_count    0
dtype: int64

# Splitting of data

Mobility Data Files
- uid: User ID 
- d: Current date 
- t: Specific time of the record 
- x: x-axis coordinate (grid column number, counted from left to right, starting from 1) 
- y: y-axis coordinate (grid row number, counted from top to bottom, starting from 1)

POI Distribution Data
- x, y: The x and y-axis coordinates of the grid 
- category: POI category 
- POI_count: The number of that POI category present in the grid 

In [9]:
POIdata_cityA['xy'] = POIdata_cityA['x'].apply(lambda x: f"{x:03d}") + POIdata_cityA['y'].apply(lambda y: f"{y:03d}")
POIdata_cityB['xy'] = POIdata_cityB['x'].apply(lambda x: f"{x:03d}") + POIdata_cityB['y'].apply(lambda y: f"{y:03d}")
POIdata_cityC['xy'] = POIdata_cityC['x'].apply(lambda x: f"{x:03d}") + POIdata_cityC['y'].apply(lambda y: f"{y:03d}")
POIdata_cityD['xy'] = POIdata_cityD['x'].apply(lambda x: f"{x:03d}") + POIdata_cityD['y'].apply(lambda y: f"{y:03d}")

POIdata_cityA.head()

x  y  category  POI_count      xy
0  1  1        74          4  001001
1  1  1        48          4  001001
2  1  1        79          2  001001
3  1  1        69          2  001001
4  1  1        73          1  001001

In [10]:
basket_cityA = POIdata_cityA.groupby(['xy', 'category'])['POI_count'].sum().unstack().reset_index().fillna(0).set_index(['xy'])
basket_cityB = POIdata_cityB.groupby(['xy', 'category'])['POI_count'].sum().unstack().reset_index().fillna(0).set_index(['xy'])
basket_cityC = POIdata_cityC.groupby(['xy', 'category'])['POI_count'].sum().unstack().reset_index().fillna(0).set_index(['xy'])
basket_cityD = POIdata_cityD.groupby(['xy', 'category'])['POI_count'].sum().unstack().reset_index().fillna(0).set_index(['xy'])

basket_cityA.head()

category   1    2    3    4    5    6    7    8    9    10  ...   76   77  \
xy                                                          ...             
001001    0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0   
001002    0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  1.0  0.0   
001003    0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0   
001004    0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0   
001005    0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  1.0  0.0   

category   78   79   80   81   82   83   84   85  
xy                                                
001001    0.0  2.0  0.0  0.0  0.0  0.0  0.0  0.0  
001002    0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  
001003    0.0  0.0  0.0  2.0  2.0  0.0  0.0  0.0  
001004    0.0  1.0  0.0  2.0  0.0  0.0  0.0  0.0  
001005    0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  

[5 rows x 85 columns]

In [11]:
basket_cityA = basket_cityA.astype(bool)
basket_cityB = basket_cityB.astype(bool)
basket_cityC = basket_cityC.astype(bool)
basket_cityD = basket_cityD.astype(bool)

basket_cityA.head()

category     1      2      3      4      5      6      7      8      9   \
xy                                                                        
001001    False  False  False  False  False  False  False  False  False   
001002    False  False  False  False  False  False  False  False  False   
001003    False  False  False  False  False  False  False  False  False   
001004    False  False  False  False  False  False  False  False  False   
001005    False  False  False  False  False  False  False  False  False   

category     10  ...     76     77     78     79     80     81     82     83  \
xy               ...                                                           
001001    False  ...  False  False  False   True  False  False  False  False   
001002    False  ...   True  False  False  False   True  False  False  False   
001003    False  ...  False  False  False  False  False   True   True  False   
001004    False  ...  False  False  False   True  False   True  False  False   
001005    False  ...   True  False  False   True  False  False  False  False   

category     84     85  
xy                      
001001    False  False  
001002    False  False  
001003    False  False  
001004    False  False  
001005    False  False  

[5 rows x 85 columns]

In [12]:
category_names = {i: POI_datacategories.iloc[i-1, 0] for i in range(1, 86)}

basket_cityA.columns = [category_names[col] for col in basket_cityA.columns]
basket_cityB.columns = [category_names[col] for col in basket_cityB.columns]
basket_cityC.columns = [category_names[col] for col in basket_cityC.columns]
basket_cityD.columns = [category_names[col] for col in basket_cityD.columns]

basket_cityA.head()

Food  Shopping  Entertainment  Japanese restaurant  \
xy                                                            
001001  False     False          False                False   
001002  False     False          False                False   
001003  False     False          False                False   
001004  False     False          False                False   
001005  False     False          False                False   

        Western restaurant  Eat all you can restaurant  Chinese restaurant  \
xy                                                                           
001001               False                       False               False   
001002               False                       False               False   
001003               False                       False               False   
001004               False                       False               False   
001005               False                       False               False   

        Indian restaurant  Ramen restaurant  Curry restaurant  ...  \
xy                                                             ...   
001001              False             False             False  ...   
001002              False             False             False  ...   
001003              False             False             False  ...   
001004              False             False             False  ...   
001005              False             False             False  ...   

        Accountant Office  IT Office  Publisher Office  Building Material  \
xy                                                                          
001001              False      False             False               True   
001002               True      False             False              False   
001003              False      False             False              False   
001004              False      False             False               True   
001005               True      False             False               True   

        Gardening  Heavy Industry    NPO  Utility Copany   Port  \
xy                                                                
001001      False           False  False           False  False   
001002       True           False  False           False  False   
001003      False            True   True           False  False   
001004      False            True  False           False  False   
001005      False           False  False           False  False   

        Research Facility  
xy                         
001001              False  
001002              False  
001003              False  
001004              False  
001005              False  

[5 rows x 85 columns]

# Building the models and analyzing the results

In [13]:
def apriori_algorithm(basket, min_support):

    # Building the model 
    frq_items = apriori(basket, min_support = min_support, use_colnames = True, low_memory=True) 

    # Collecting the inferred rules in a dataframe 
    rules = association_rules(frq_items, metric ="lift", min_threshold = 1, num_itemsets = basket.shape[1]) 
    rules = rules.sort_values(['confidence', 'lift'], ascending =[False, False]) 

    return rules

In [14]:
rules_cityA = apriori_algorithm(basket_cityA, 0.06)
rules_cityA.head()

antecedents   consequents  \
3296614  (Café, Hospital, Transit Station, Accountant O...  (Hair Salon)   
796122   (Café, Hospital, Clothes Store, Japanese resta...  (Hair Salon)   
2462421  (Café, Hospital, Transit Station, Fishing, Lau...  (Hair Salon)   
4066489  (Café, Hospital, Transit Station, Japanese res...  (Hair Salon)   
788125   (Japanese restaurant, Real Estate, Fishing, La...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
3296614            0.060856            0.374814  0.060111    0.987765   
796122             0.063089            0.374814  0.062295    0.987411   
2462421            0.062196            0.374814  0.061402    0.987231   
4066489            0.061451            0.374814  0.060657    0.987076   
788125             0.060905            0.374814  0.060111    0.986960   

             lift  representativity  leverage  conviction  zhangs_metric  \
3296614  2.635348               1.0  0.037302   51.098547       0.660754   
796122   2.634405               1.0  0.038648   49.663224       0.662184   
2462421  2.633922               1.0  0.038090   48.959890       0.661479   
4066489  2.633510               1.0  0.037624   48.373778       0.660891   
788125   2.633201               1.0  0.037283   47.943962       0.660460   

          jaccard  certainty  kulczynski  
3296614  0.160058   0.980430    0.574071  
796122   0.165852   0.979864    0.576807  
2462421  0.163473   0.979575    0.575525  
4066489  0.161491   0.979328    0.574454  
788125   0.160037   0.979142    0.573668

In [15]:
rules_cityB = apriori_algorithm(basket_cityB, 0.03)
rules_cityB.head()

antecedents   consequents  \
393488   (Home Appliances, Accountant Office, Post Offi...  (Hair Salon)   
393548   (Heavy Industry, Home Appliances, Post Office,...  (Hair Salon)   
867646   (Café, Japanese restaurant, Heavy Industry, Re...  (Hair Salon)   
982285   (Cram School, Interior Shop, Accountant Office...  (Hair Salon)   
1030769  (Japanese restaurant, Heavy Industry, Real Est...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
393488             0.030688              0.1895  0.030688         1.0   
393548             0.030579              0.1895  0.030579         1.0   
867646             0.031127              0.1895  0.031127         1.0   
982285             0.030688              0.1895  0.030688         1.0   
1030769            0.031565              0.1895  0.031565         1.0   

             lift  representativity  leverage  conviction  zhangs_metric  \
393488   5.277039               1.0  0.024873         inf       0.836160   
393548   5.277039               1.0  0.024784         inf       0.836066   
867646   5.277039               1.0  0.025228         inf       0.836538   
982285   5.277039               1.0  0.024873         inf       0.836160   
1030769  5.277039               1.0  0.025584         inf       0.836917   

          jaccard  certainty  kulczynski  
393488   0.161943        1.0    0.580972  
393548   0.161365        1.0    0.580682  
867646   0.164257        1.0    0.582128  
982285   0.161943        1.0    0.580972  
1030769  0.166570        1.0    0.583285

In [16]:
rules_cityC = apriori_algorithm(basket_cityC, 0.09)
rules_cityC.head()

antecedents    consequents  \
3951142  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4696044  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
3911771  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   
2846234  (Elderly Care Home, Bank, Home Appliances, Acc...  (Real Estate)   
4628213  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   

         antecedent support  consequent support   support  confidence  \
3951142            0.093817            0.396186  0.093817    1.000000   
4696044            0.092279            0.396186  0.092279    1.000000   
3911771            0.102122            0.396186  0.101815    0.996988   
2846234            0.100277            0.396186  0.099969    0.996933   
4628213            0.100277            0.396186  0.099969    0.996933   

             lift  representativity  leverage  conviction  zhangs_metric  \
3951142  2.524068               1.0  0.056648         inf       0.666327   
4696044  2.524068               1.0  0.055720         inf       0.665198   
3911771  2.516466               1.0  0.061355  200.466318       0.671158   
2846234  2.516326               1.0  0.060241  196.843433       0.669756   
4628213  2.516326               1.0  0.060241  196.843433       0.669756   

          jaccard  certainty  kulczynski  
3951142  0.236801   1.000000    0.618401  
4696044  0.232919   1.000000    0.616460  
3911771  0.256788   0.995012    0.626988  
2846234  0.252133   0.994920    0.624631  
4628213  0.252133   0.994920    0.624631

In [17]:
rules_cityD = apriori_algorithm(basket_cityD, 0.03)
rules_cityD.head()

antecedents   consequents  \
711730  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
941177  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
941685  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
943463  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
713368  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   

        antecedent support  consequent support   support  confidence  \
711730            0.032490            0.230251  0.032308    0.994398   
941177            0.030943            0.230251  0.030761    0.994118   
941685            0.031216            0.230251  0.030943    0.991254   
943463            0.030397            0.230251  0.030124    0.991018   
713368            0.032763            0.230251  0.032399    0.988889   

            lift  representativity  leverage  conviction  zhangs_metric  \
711730  4.318752               1.0  0.024827  137.400164       0.794257   
941177  4.317535               1.0  0.023636  130.857299       0.792922   
941685  4.305097               1.0  0.023755   88.007948       0.792454   
943463  4.304073               1.0  0.023125   85.698702       0.791728   
713368  4.294827               1.0  0.024855   69.277394       0.793148   

         jaccard  certainty  kulczynski  
711730  0.140205   0.992722    0.567357  
941177  0.133491   0.992358    0.563857  
941685  0.134228   0.988637    0.562820  
943463  0.130675   0.988331    0.560924  
713368  0.140489   0.985565    0.564800

# Analysis of rule associations

## 1. Analysis of City A

In [18]:
cityA_analysis = rules_cityA.iloc[:,0:7]
cityA_analysis['interest'] = cityA_analysis['confidence'] - cityA_analysis['consequent support']
cityA_analysis.to_csv('cityA.csv', index=False)

cityA_analysis.head()

antecedents   consequents  \
3296614  (Café, Hospital, Transit Station, Accountant O...  (Hair Salon)   
796122   (Café, Hospital, Clothes Store, Japanese resta...  (Hair Salon)   
2462421  (Café, Hospital, Transit Station, Fishing, Lau...  (Hair Salon)   
4066489  (Café, Hospital, Transit Station, Japanese res...  (Hair Salon)   
788125   (Japanese restaurant, Real Estate, Fishing, La...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
3296614            0.060856            0.374814  0.060111    0.987765   
796122             0.063089            0.374814  0.062295    0.987411   
2462421            0.062196            0.374814  0.061402    0.987231   
4066489            0.061451            0.374814  0.060657    0.987076   
788125             0.060905            0.374814  0.060111    0.986960   

             lift  interest  
3296614  2.635348  0.612951  
796122   2.634405  0.612598  
2462421  2.633922  0.612417  
4066489  2.633510  0.612262  
788125   2.633201  0.612146

In [19]:
# Analyse the association rules based on the support value
cityA_analysis.sort_values('support', ascending = False).head(20)

antecedents          consequents  antecedent support  \
1283  (Building Material)     (Heavy Industry)            0.453291   
1282     (Heavy Industry)  (Building Material)            0.559069   
1090     (Heavy Industry)    (Home Appliances)            0.559069   
1091    (Home Appliances)     (Heavy Industry)            0.458255   
1089  (Building Material)    (Home Appliances)            0.453291   
1088    (Home Appliances)  (Building Material)            0.458255   
1059        (Real Estate)     (Heavy Industry)            0.372679   
1058     (Heavy Industry)        (Real Estate)            0.559069   
1216     (Heavy Industry)         (Hair Salon)            0.559069   
1217         (Hair Salon)     (Heavy Industry)            0.374814   
1057  (Building Material)        (Real Estate)            0.453291   
1056        (Real Estate)  (Building Material)            0.372679   
759     (Transit Station)     (Heavy Industry)            0.406979   
758      (Heavy Industry)    (Transit Station)            0.559069   
1214         (Hair Salon)  (Building Material)            0.374814   
1215  (Building Material)         (Hair Salon)            0.453291   
1031    (Home Appliances)        (Real Estate)            0.458255   
1030        (Real Estate)    (Home Appliances)            0.372679   
1042        (Real Estate)         (Hair Salon)            0.372679   
1043         (Hair Salon)        (Real Estate)            0.374814   

      consequent support   support  confidence      lift  interest  
1283            0.559069  0.335104    0.739269  1.322321  0.180200  
1282            0.453291  0.335104    0.599396  1.322321  0.146105  
1090            0.458255  0.331927    0.593714  1.295598  0.135459  
1091            0.559069  0.331927    0.724328  1.295598  0.165260  
1089            0.458255  0.301747    0.665681  1.452644  0.207426  
1088            0.453291  0.301747    0.658471  1.452644  0.205180  
1059            0.559069  0.292316    0.784363  1.402982  0.225295  
1058            0.372679  0.292316    0.522862  1.402982  0.150183  
1216            0.374814  0.286062    0.511675  1.365145  0.136862  
1217            0.559069  0.286062    0.763210  1.365145  0.204141  
1057            0.372679  0.277921    0.613119  1.645164  0.240439  
1056            0.453291  0.277921    0.745738  1.645164  0.292447  
759             0.559069  0.272163    0.668740  1.196168  0.109671  
758             0.406979  0.272163    0.486815  1.196168  0.079836  
1214            0.453291  0.272014    0.725732  1.601028  0.272441  
1215            0.374814  0.272014    0.600088  1.601028  0.225274  
1031            0.372679  0.268291    0.585464  1.570958  0.212784  
1030            0.458255  0.268291    0.719899  1.570958  0.261644  
1042            0.374814  0.266951    0.716303  1.911089  0.341489  
1043            0.372679  0.266951    0.712224  1.911089  0.339544

Range of top 20 support values: 0.335104 to 0.266951	

Observations: 
- All of the rules with the top 20 support values consists of a single antecedent and consequent (X -> Y), which is not surprising.

In [20]:
# Analyse the association rules based on the confidence value
cityA_analysis.sort_values('confidence', ascending = False).head(20)

antecedents   consequents  \
3296614  (Café, Hospital, Transit Station, Accountant O...  (Hair Salon)   
796122   (Café, Hospital, Clothes Store, Japanese resta...  (Hair Salon)   
2462421  (Café, Hospital, Transit Station, Fishing, Lau...  (Hair Salon)   
4066489  (Café, Hospital, Transit Station, Japanese res...  (Hair Salon)   
788125   (Japanese restaurant, Real Estate, Fishing, La...  (Hair Salon)   
1147600  (Café, Hospital, Clothes Store, Real Estate, F...  (Hair Salon)   
2024193  (Café, Bank, Hospital, Japanese restaurant, La...  (Hair Salon)   
2043848  (Café, Chiropractic, Japanese restaurant, Fish...  (Hair Salon)   
1147972  (Café, Hospital, Clothes Store, Accountant Off...  (Hair Salon)   
2365906  (Bank, Cram School, Japanese restaurant, Real ...  (Hair Salon)   
2021799  (Café, Hospital, Japanese restaurant, Real Est...  (Hair Salon)   
3293566  (Café, Hospital, Transit Station, Japanese res...  (Hair Salon)   
520258   (Clothes Store, Hospital, Accountant Office, C...  (Hair Salon)   
2415927  (Café, Hospital, Interior Shop, Real Estate, F...  (Hair Salon)   
2285266  (Bank, Hospital, Japanese restaurant, Fishing,...  (Hair Salon)   
2112142  (Hospital, Clothes Store, Accountant Office, J...  (Hair Salon)   
1460701  (Hospital, Clothes Store, Accountant Office, H...  (Hair Salon)   
1280404       (Café, Bank, Cram School, Fishing, Laundry )  (Hair Salon)   
2043471  (Café, Japanese restaurant, Real Estate, Fishi...  (Hair Salon)   
901709   (Hospital, Clothes Store, Japanese restaurant,...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
3296614            0.060856            0.374814  0.060111    0.987765   
796122             0.063089            0.374814  0.062295    0.987411   
2462421            0.062196            0.374814  0.061402    0.987231   
4066489            0.061451            0.374814  0.060657    0.987076   
788125             0.060905            0.374814  0.060111    0.986960   
1147600            0.060806            0.374814  0.060012    0.986939   
2024193            0.060806            0.374814  0.060012    0.986939   
2043848            0.060806            0.374814  0.060012    0.986939   
1147972            0.062593            0.374814  0.061749    0.986519   
2365906            0.062295            0.374814  0.061451    0.986454   
2021799            0.062246            0.374814  0.061402    0.986443   
3293566            0.062246            0.374814  0.061402    0.986443   
520258             0.062196            0.374814  0.061352    0.986433   
2415927            0.062196            0.374814  0.061352    0.986433   
2285266            0.061997            0.374814  0.061154    0.986389   
2112142            0.061948            0.374814  0.061104    0.986378   
1460701            0.061898            0.374814  0.061054    0.986367   
1280404            0.061600            0.374814  0.060756    0.986301   
2043471            0.064181            0.374814  0.063288    0.986079   
901709             0.063834            0.374814  0.062941    0.986003   

             lift  interest  
3296614  2.635348  0.612951  
796122   2.634405  0.612598  
2462421  2.633922  0.612417  
4066489  2.633510  0.612262  
788125   2.633201  0.612146  
1147600  2.633144  0.612125  
2024193  2.633144  0.612125  
2043848  2.633144  0.612125  
1147972  2.632023  0.611705  
2365906  2.631851  0.611640  
2021799  2.631822  0.611630  
3293566  2.631822  0.611630  
520258   2.631793  0.611619  
2415927  2.631793  0.611619  
2285266  2.631677  0.611575  
2112142  2.631648  0.611564  
1460701  2.631619  0.611553  
1280404  2.631443  0.611488  
2043471  2.630850  0.611265  
901709   2.630647  0.611189

Range of top 20 confidence values: 0.987765 to 0.987765

Observations: 
- Based on the top 20 confidence values, all of the rules have (Hair Salon) as the sole consequent.

- As for the antecedents, there were some items that commonly appeared in the top 20 rules:
    - Chiropractic
    - Japanese restaurant
    - Fishing
    - Hospital
    - Café
    - Clothes Store

- By intution, these items should have a strong relation as they are related to health and leisure, which could explain the high frequency of occurance.

In [21]:
# Analyse the association rules based on the lift value
cityA_analysis.sort_values('lift', ascending = False).head(20)

antecedents  \
4073989  (Hospital, Transit Station, Japanese restauran...   
4073764  (Café, Hair Salon, Bank, Heavy Industry, Real ...   
4073913  (Hospital, Transit Station, Café, Accountant O...   
4073840  (Hair Salon, Bank, Japanese restaurant, Heavy ...   
4109512  (Hair Salon, Transit Station, Japanese restaur...   
4109641      (Hospital, Bank, Laundry , Accountant Office)   
4073994  (Real Estate, Hospital, Japanese restaurant, A...   
4073759  (Café, Hair Salon, Bank, Transit Station, Heav...   
4073769  (Café, Hair Salon, Hospital, Transit Station, ...   
4073984  (Real Estate, Bank, Japanese restaurant, Accou...   
4109694  (Real Estate, Transit Station, Fishing, Japane...   
4109459  (Hair Salon, Hospital, Bank, Accountant Office...   
4109535  (Hospital, Bank, Accountant Office, Real Estat...   
4109618  (Hair Salon, Transit Station, Fishing, Japanes...   
4073917   (Real Estate, Hospital, Café, Accountant Office)   
4073836  (Hair Salon, Bank, Transit Station, Japanese r...   
4073943  (Hair Salon, Bank, Japanese restaurant, Accoun...   
4073810  (Café, Hospital, Transit Station, Heavy Indust...   
4273501  (Café, Hospital, Heavy Industry, Real Estate, ...   
4273584  (Hair Salon, Accountant Office, Transit Statio...   

                                               consequents  \
4073989  (Café, Hair Salon, Bank, Heavy Industry, Real ...   
4073764  (Hospital, Transit Station, Japanese restauran...   
4073913  (Hair Salon, Bank, Japanese restaurant, Heavy ...   
4073840  (Hospital, Transit Station, Café, Accountant O...   
4109512      (Hospital, Bank, Laundry , Accountant Office)   
4109641  (Hair Salon, Transit Station, Japanese restaur...   
4073994  (Café, Hair Salon, Bank, Transit Station, Heav...   
4073759  (Real Estate, Hospital, Japanese restaurant, A...   
4073769  (Real Estate, Bank, Japanese restaurant, Accou...   
4073984  (Café, Hair Salon, Hospital, Transit Station, ...   
4109694  (Hair Salon, Hospital, Bank, Accountant Office...   
4109459  (Real Estate, Transit Station, Fishing, Japane...   
4109535  (Hair Salon, Transit Station, Fishing, Japanes...   
4109618  (Hospital, Bank, Accountant Office, Real Estat...   
4073917  (Hair Salon, Bank, Transit Station, Japanese r...   
4073836   (Real Estate, Hospital, Café, Accountant Office)   
4073943  (Café, Hospital, Transit Station, Heavy Indust...   
4073810  (Hair Salon, Bank, Japanese restaurant, Accoun...   
4273501  (Hair Salon, Accountant Office, Transit Statio...   
4273584  (Café, Hospital, Heavy Industry, Real Estate, ...   

         antecedent support  consequent support   support  confidence  \
4073989            0.100665            0.099772  0.060111    0.597140   
4073764            0.099772            0.100665  0.060111    0.602488   
4073913            0.097389            0.103594  0.060111    0.617227   
4073840            0.103594            0.097389  0.060111    0.580259   
4109512            0.097786            0.104487  0.060558    0.619289   
4109641            0.104487            0.097786  0.060558    0.579572   
4073994            0.110196            0.092078  0.060111    0.545495   
4073759            0.092078            0.110196  0.060111    0.652830   
4073769            0.098829            0.102700  0.060111    0.608237   
4073984            0.102700            0.098829  0.060111    0.585307   
4109694            0.102750            0.099523  0.060558    0.589372   
4109459            0.099523            0.102750  0.060558    0.608479   
4109535            0.099027            0.103296  0.060558    0.611529   
4109618            0.103296            0.099027  0.060558    0.586257   
4073917            0.105927            0.095900  0.060111    0.567479   
4073836            0.095900            0.105927  0.060111    0.626812   
4073943            0.103296            0.098382  0.060111    0.581932   
4073810            0.098382            0.103296  0.060111    0.610999   
4273501            0.100814            0.102254  0.060

Range of top 20 lift values: 5.985066 to 5.913030

Observations: 
- This implies that these groups of items seemingly have quite a significant impact on each other, but to get more information beyond this surface level interpretation would require more detailed analysis as the antecedents and consequents are quite varied.

In [38]:
# Analyse the association rules based on the interest value
cityA_analysis.sort_values('interest', ascending = False).head(20)

antecedents  \
4109424  (Hospital, Bank, Transit Station, Accountant O...   
4105854  (Hospital, Home Appliances, Transit Station, A...   
4275226  (Hospital, Transit Station, Accountant Office,...   
4110957  (Hospital, Transit Station, Accountant Office,...   
4111469  (Hospital, Transit Station, Accountant Office,...   
3415301  (Hospital, Transit Station, Japanese restauran...   
4152779  (Hospital, Bank, Accountant Office, Japanese r...   
4152267  (Hospital, Bank, Accountant Office, Japanese r...   
2640721  (Hair Salon, Hospital, Transit Station, Fishin...   
4066568  (Café, Hospital, Transit Station, Japanese res...   
3293618  (Café, Hospital, Transit Station, Japanese res...   
4103305  (Building Material, Hospital, Transit Station,...   
3516139  (Hospital, Accountant Office, Japanese restaur...   
3466355  (Accountant Office, Transit Station, Japanese ...   
4103820  (Hospital, Transit Station, Accountant Office,...   
3466101  (Home Appliances, Transit Station, Japanese re...   
4105762  (Hair Salon, Hospital, Home Appliances, Transi...   
3383294  (Hospital, Park, Transit Station, Japanese res...   
4076258  (Café, Accountant Office, Transit Station, Jap...   
3334512  (Café, Hospital, Accountant Office, Japanese r...   

                                 consequents  antecedent support  \
4109424  (Real Estate, Hair Salon, Laundry )            0.066812   
4105854  (Real Estate, Hair Salon, Laundry )            0.067209   
4275226  (Real Estate, Hair Salon, Laundry )            0.067060   
4110957  (Real Estate, Hair Salon, Laundry )            0.071379   
4111469  (Real Estate, Hair Salon, Laundry )            0.072074   
3415301  (Real Estate, Hair Salon, Laundry )            0.069542   
4152779  (Real Estate, Hair Salon, Laundry )            0.067855   
4152267  (Real Estate, Hair Salon, Laundry )            0.067805   
2640721                  (Real Estate, Bank)            0.068103   
4066568  (Real Estate, Hair Salon, Laundry )            0.067557   
3293618  (Real Estate, Hair Salon, Laundry )            0.068401   
4103305  (Real Estate, Hair Salon, Laundry )            0.068500   
3516139  (Real Estate, Hair Salon, Laundry )            0.067954   
3466355  (Real Estate, Hair Salon, Laundry )            0.068103   
4103820  (Real Estate, Hair Salon, Laundry )            0.069195   
3466101  (Real Estate, Hair Salon, Laundry )            0.067011   
4105762              (Real Estate, Laundry )            0.065472   
3383294  (Real Estate, Hair Salon, Laundry )            0.070188   
4076258  (Real Estate, Hair Salon, Laundry )            0.067557   
3334512  (Real Estate, Hair Salon, Laundry )            0.067458   

         consequent support   support  confidence      lift  interest  
4109424            0.197012  0.060558    0.906389  4.600685  0.709377  
4105854            0.197012  0.060856    0.905465  4.595995  0.708453  
4275226            0.197012  0.060657    0.904515  4.591172  0.707503  
4110957            0.197012  0.064281    0.900556  4.571078  0.703545  
4111469            0.197012  0.064876    0.900138  4.568953  0.703126  
3415301            0.197012  0.062543    0.899358  4.564993  0.702346  
4152779            0.197012  0.060955    0.898317  4.559714  0.701306  
4152267            0.197012  0.060905    0.898243  4.559336  0.701231  
2640721            0.188822  0.060608    0.889942  4.713135  0.701120  
4066568            0.197012  0.060657    0.897869  4.557438  0.700857  
3293618            0.197012  0.061402    0.897678  4.556467  0.700666  
4103305            0.197012  0.061451    0.897101  4.553541  0.700090  
3516139            0.197012  0.060955    0.897005  4.553052  0.699993  
3466355            0.197012  0.061054    0.896501  4.550496  0.699490  
4103820            0.197012  0.061997    0.895983  4.547863  0.698971  
3466101            0.197012  0.060012    0.895556  4.545695  0.698544  
4105762            0.231014  0.060856    0.929492  4.023538  0.698478  
3383294            

Range of top 20 interest values: 0.709377 to 0.697028

Observations: 
- Surprisingly, majority of the consequents consist of the group (Real Estate, Hair Salon, Laundry ). This implies that these items have a meaningful association with many of the items, as the antecedents consists of a mix of different items for the top 20 rules.

## 2. Analysis of City B

In [23]:
cityB_analysis = rules_cityB.iloc[:,0:7]
cityB_analysis['interest'] = cityB_analysis['confidence'] - cityB_analysis['consequent support']
cityB_analysis.to_csv('cityB.csv', index=False)

cityB_analysis.head()

antecedents   consequents  \
393488   (Home Appliances, Accountant Office, Post Offi...  (Hair Salon)   
393548   (Heavy Industry, Home Appliances, Post Office,...  (Hair Salon)   
867646   (Café, Japanese restaurant, Heavy Industry, Re...  (Hair Salon)   
982285   (Cram School, Interior Shop, Accountant Office...  (Hair Salon)   
1030769  (Japanese restaurant, Heavy Industry, Real Est...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
393488             0.030688              0.1895  0.030688         1.0   
393548             0.030579              0.1895  0.030579         1.0   
867646             0.031127              0.1895  0.031127         1.0   
982285             0.030688              0.1895  0.030688         1.0   
1030769            0.031565              0.1895  0.031565         1.0   

             lift  interest  
393488   5.277039    0.8105  
393548   5.277039    0.8105  
867646   5.277039    0.8105  
982285   5.277039    0.8105  
1030769  5.277039    0.8105

In [24]:
# Analyse the association rules based on the support value
cityB_analysis.sort_values('support', ascending = False).head(20)

antecedents          consequents  antecedent support  \
709   (Building Material)    (Transit Station)            0.270934   
708     (Transit Station)  (Building Material)            0.359711   
711     (Transit Station)     (Heavy Industry)            0.359711   
710      (Heavy Industry)    (Transit Station)            0.303924   
683     (Home Appliances)    (Transit Station)            0.269838   
682     (Transit Station)    (Home Appliances)            0.359711   
1042    (Home Appliances)  (Building Material)            0.269838   
1043  (Building Material)    (Home Appliances)            0.270934   
1230     (Heavy Industry)  (Building Material)            0.303924   
1231  (Building Material)     (Heavy Industry)            0.270934   
1044     (Heavy Industry)    (Home Appliances)            0.303924   
1045    (Home Appliances)     (Heavy Industry)            0.269838   
702     (Transit Station)             (Church)            0.359711   
703              (Church)    (Transit Station)            0.375055   
696          (Hair Salon)    (Transit Station)            0.189500   
697     (Transit Station)         (Hair Salon)            0.359711   
700     (Transit Station)   (Community Center)            0.359711   
701    (Community Center)    (Transit Station)            0.253836   
706   (Accountant Office)    (Transit Station)            0.201666   
707     (Transit Station)  (Accountant Office)            0.359711   

      consequent support   support  confidence      lift  interest  
709             0.359711  0.162429    0.599515  1.666658  0.239804  
708             0.270934  0.162429    0.451554  1.666658  0.180620  
711             0.303924  0.160346    0.445765  1.466700  0.141841  
710             0.359711  0.160346    0.527587  1.466700  0.167877  
683             0.359711  0.153113    0.567425  1.577448  0.207714  
682             0.269838  0.153113    0.425655  1.577448  0.155817  
1042            0.270934  0.152784    0.566206  2.089833  0.295273  
1043            0.269838  0.152784    0.563916  2.089833  0.294078  
1230            0.270934  0.151249    0.497656  1.836818  0.226722  
1231            0.303924  0.151249    0.558252  1.836818  0.254329  
1044            0.269838  0.145989    0.480346  1.780129  0.210508  
1045            0.303924  0.145989    0.541024  1.780129  0.237100  
702             0.375055  0.142043    0.394881  1.052863  0.019826  
703             0.359711  0.142043    0.378726  1.052863  0.019015  
696             0.359711  0.136673    0.721226  2.005017  0.361515  
697             0.189500  0.136673    0.379951  2.005017  0.190451  
700             0.253836  0.135028    0.375381  1.478832  0.121545  
701             0.359711  0.135028    0.531952  1.478832  0.172241  
706             0.359711  0.129220    0.640761  1.781323  0.281050  
707             0.201666  0.129220    0.359232  1.781323  0.157566

Range of top 20 support values: 0.162429 to 0.129220

Observations: 
- The support value in City B is quite low, suggesting that the rules are quite infrequent on a whole.
- As stated before, single antecedent and consequent pair is observed.

In [25]:
# Analyse the association rules based on the confidence value
cityB_analysis.sort_values('confidence', ascending = False).head(20)

antecedents   consequents  \
393488   (Home Appliances, Accountant Office, Post Offi...  (Hair Salon)   
4781583  (Hospital, Interior Shop, Accountant Office, T...  (Hair Salon)   
5654581  (Bank, Hospital, Interior Shop, Accountant Off...  (Hair Salon)   
5566189  (Hospital, Bank, Interior Shop, Accountant Off...  (Hair Salon)   
5546123  (Hospital, Bank, Interior Shop, Transit Statio...  (Hair Salon)   
5545615  (Hospital, Bank, Interior Shop, Accountant Off...  (Hair Salon)   
5533677  (Hospital, Bank, Park, Interior Shop, Accounta...  (Hair Salon)   
5384579  (Bank, Hospital, Accountant Office, Real Estat...  (Hair Salon)   
5353591  (Bank, Hospital, Transit Station, Accountant O...  (Hair Salon)   
5275867  (Hospital, Bank, Post Office, Japanese restaur...  (Hair Salon)   
5275359  (Hospital, Post Office, Japanese restaurant, H...  (Hair Salon)   
5274851  (Hospital, Post Office, Accountant Office, Jap...  (Hair Salon)   
5274597  (Hospital, Bank, Post Office, Japanese restaur...  (Hair Salon)   
5274089  (Bank, Hospital, Post Office, Accountant Offic...  (Hair Salon)   
5262913  (Hospital, Cram School, Japanese restaurant, H...  (Hair Salon)   
5262659  (Hospital, Cram School, Accountant Office, Jap...  (Hair Salon)   
5262405  (Hospital, Cram School, Accountant Office, Jap...  (Hair Salon)   
5262151  (Hospital, Bank, Cram School, Japanese restaur...  (Hair Salon)   
5261643  (Bank, Hospital, Cram School, Accountant Offic...  (Hair Salon)   
5260881  (Hospital, Cram School, Home Appliances, Accou...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
393488             0.030688              0.1895  0.030688         1.0   
4781583            0.030688              0.1895  0.030688         1.0   
5654581            0.031565              0.1895  0.031565         1.0   
5566189            0.032990              0.1895  0.032990         1.0   
5546123            0.030798              0.1895  0.030798         1.0   
5545615            0.031017              0.1895  0.031017         1.0   
5533677            0.030469              0.1895  0.030469         1.0   
5384579            0.030798              0.1895  0.030798         1.0   
5353591            0.030140              0.1895  0.030140         1.0   
5275867            0.030031              0.1895  0.030031         1.0   
5275359            0.030579              0.1895  0.030579         1.0   
5274851            0.030907              0.1895  0.030907         1.0   
5274597            0.030907              0.1895  0.030907         1.0   
5274089            0.031236              0.1895  0.031236         1.0   
5262913            0.030907              0.1895  0.030907         1.0   
5262659            0.030798              0.1895  0.030798         1.0   
5262405            0.031236              0.1895  0.031236         1.0   
5262151            0.030031              0.1895  0.030031         1.0   
5261643            0.030907              0.1895  0.030907         1.0   
5260881            0.030140              0.1895  0.030140         1.0   

             lift  interest  
393488   5.277039    0.8105  
4781583  5.277039    0.8105  
5654581  5.277039    0.8105  
5566189  5.277039    0.8105  
5546123  5.277039    0.8105  
5545615  5.277039    0.8105  
5533677  5.277039    0.8105  
5384579  5.277039    0.8105  
5353591  5.277039    0.8105  
5275867  5.277039    0.8105  
5275359  5.277039    0.8105  
5274851  5.277039    0.8105  
5274597  5.277039    0.8105  
5274089  5.277039    0.8105  
5262913  5.277039    0.8105  
5262659  5.277039    0.8105  
5262405  5.277039    0.8105  
5262151  5.277039    0.8105  
5261643  5.277039    0.8105  
5260881  5.277039    0.8105

Range of top 20 confidence values: 1.0 to 1.0

Observations: 
- Like before, (Hair Salon) is the sole consequent for the top 20 confidence values.
- The confidence is strangely high, indicating that hair salon is always visited for each group of antecedents in the top 20 confidence values. This might be due to the low min_support chosen.

In [26]:
# Bnalyse the association rules based on the lift value
cityB_analysis.sort_values('lift', ascending = False).head(20)

antecedents  \
7585108  (Hospital, Interior Shop, Real Estate, Laundry...   
7584937  (Hair Salon, Bank, Accountant Office, Heavy In...   
7585095  (Hospital, Interior Shop, Accountant Office, R...   
7584950  (Hair Salon, Bank, Heavy Industry, Fishing, Bu...   
7584749  (Hair Salon, Bank, Accountant Office, Heavy In...   
7585296   (Real Estate, Hospital, Interior Shop, Laundry )   
6620141  (Bank, Fishing, Japanese restaurant, Accountan...   
6619976  (Hair Salon, Hospital, Interior Shop, Real Est...   
6619959  (Hair Salon, Bank, Accountant Office, Japanese...   
6620161  (Hospital, Fishing, Japanese restaurant, Accou...   
6620158   (Real Estate, Hospital, Interior Shop, Laundry )   
6619956  (Hair Salon, Bank, Interior Shop, Real Estate,...   
7584753  (Hair Salon, Bank, Heavy Industry, Real Estate...   
7585292  (Hospital, Interior Shop, Laundry , Accountant...   
7585104  (Hospital, Interior Shop, Heavy Industry, Real...   
7584941  (Hair Salon, Bank, Accountant Office, Fishing,...   
6620154  (Real Estate, Hospital, Interior Shop, Japanes...   
6619963  (Hair Salon, Bank, Accountant Office, Fishing,...   
6620072     (Hair Salon, Bank, Fishing, Accountant Office)   
6620045  (Hospital, Interior Shop, Japanese restaurant,...   

                                               consequents  \
7585108  (Hair Salon, Bank, Accountant Office, Heavy In...   
7584937  (Hospital, Interior Shop, Real Estate, Laundry...   
7585095  (Hair Salon, Bank, Heavy Industry, Fishing, Bu...   
7584950  (Hospital, Interior Shop, Accountant Office, R...   
7584749   (Real Estate, Hospital, Interior Shop, Laundry )   
7585296  (Hair Salon, Bank, Accountant Office, Heavy In...   
6620141  (Hair Salon, Hospital, Interior Shop, Real Est...   
6619976  (Bank, Fishing, Japanese restaurant, Accountan...   
6619959   (Real Estate, Hospital, Interior Shop, Laundry )   
6620161  (Hair Salon, Bank, Interior Shop, Real Estate,...   
6620158  (Hair Salon, Bank, Accountant Office, Japanese...   
6619956  (Hospital, Fishing, Japanese restaurant, Accou...   
7584753  (Hospital, Interior Shop, Laundry , Accountant...   
7585292  (Hair Salon, Bank, Heavy Industry, Real Estate...   
7585104  (Hair Salon, Bank, Accountant Office, Fishing,...   
7584941  (Hospital, Interior Shop, Heavy Industry, Real...   
6620154  (Hair Salon, Bank, Accountant Office, Fishing,...   
6619963  (Real Estate, Hospital, Interior Shop, Japanes...   
6620072  (Hospital, Interior Shop, Japanese restaurant,...   
6620045     (Hair Salon, Bank, Fishing, Accountant Office)   

         antecedent support  consequent support   support  confidence  \
7585108            0.041868            0.043621  0.030579    0.730366   
7584937            0.043621            0.041868  0.030579    0.701005   
7585095            0.041977            0.043512  0.030579    0.728460   
7584950            0.043512            0.041977  0.030579    0.702771   
7584749            0.040224            0.045594  0.030579    0.760218   
7585296            0.045594            0.040224  0.030579    0.670673   
6620141            0.042087            0.044279  0.030798    0.731771   
6619976            0.044279            0.042087  0.030798    0.695545   
6619959            0.040881            0.045594  0.030798    0.753351   
6620161            0.040881            0.045594  0.030798    0.753351   
6620158            0.045594            0.040881  0.030798    0.675481   
6619956            0.045594            0.040881  0.030798    0.675481   
7584753            0.041100            0.045156  0.030579    0.744000   
7585292            0.045156            0.041100  0.030579    0.677184   
7585104            0.041868            0.044388  0.030579    0.730366   
7584941            0.044388            0.041868  0.030579    0.688889   
6620154            0.040991            0.045704  0.030798    0.751337   
6619963            0.045704            0.040991  0.030798    0.673861   
6620072            0.048882            0.038470  0.030

Range of top 20 lift values: 16.743377 to 16.377576

Observations: 
- No notable observation

In [27]:
# Analyse the association rules based on the interest value
cityB_analysis.sort_values('interest', ascending = False).head(20)

antecedents  \
3237741  (Bank, Post Office, Real Estate, Drug Store, L...   
5347777  (Accountant Office, Transit Station, Real Esta...   
5349300  (Bank, Accountant Office, Real Estate, Grocery...   
1372218          (Drug Store, Bank, Cram School, Laundry )   
4710745  (Accountant Office, Japanese restaurant, Heavy...   
880248   (Drug Store, Accountant Office, Japanese resta...   
6550487  (Bank, Japanese restaurant, Heavy Industry, Re...   
3167559  (Accountant Office, Real Estate, Grocery Store...   
4710999  (Japanese restaurant, Heavy Industry, Real Est...   
3167811  (Heavy Industry, Real Estate, Grocery Store, L...   
3225267  (Accountant Office, Real Estate, Drug Store, L...   
3180159  (Accountant Office, Transit Station, Drug Stor...   
3168693  (Accountant Office, Heavy Industry, Real Estat...   
7564125  (Bank, Transit Station, Accountant Office, Hea...   
4711760  (Bank, Accountant Office, Japanese restaurant,...   
4693727  (Accountant Office, Transit Station, Japanese ...   
6549468  (Transit Station, Japanese restaurant, Heavy I...   
2460693  (Driving School, Transit Station, Japanese res...   
1372466  (Drug Store, Bank, Cram School, Building Mater...   
1372404  (Heavy Industry, Drug Store, Cram School, Laun...   

                    consequents  antecedent support  consequent support  \
3237741  (Hair Salon, Hospital)            0.030907            0.092942   
5347777  (Hair Salon, Hospital)            0.030688            0.092942   
5349300  (Hair Salon, Hospital)            0.030579            0.092942   
1372218  (Hair Salon, Hospital)            0.032004            0.092942   
4710745  (Hair Salon, Hospital)            0.031675            0.092942   
880248   (Hair Salon, Hospital)            0.030907            0.092942   
6550487  (Hair Salon, Hospital)            0.030688            0.092942   
3167559  (Hair Salon, Hospital)            0.032442            0.092942   
4710999  (Hair Salon, Hospital)            0.032442            0.092942   
3167811  (Hair Salon, Hospital)            0.032113            0.092942   
3225267  (Hair Salon, Hospital)            0.031784            0.092942   
3180159  (Hair Salon, Hospital)            0.031784            0.092942   
3168693  (Hair Salon, Hospital)            0.031675            0.092942   
7564125  (Hair Salon, Hospital)            0.031565            0.092942   
4711760  (Hair Salon, Hospital)            0.031565            0.092942   
4693727  (Hair Salon, Hospital)            0.031346            0.092942   
6549468  (Hair Salon, Hospital)            0.030907            0.092942   
2460693  (Hair Salon, Hospital)            0.030907            0.092942   
1372466  (Hair Salon, Hospital)            0.030907            0.092942   
1372404  (Hair Salon, Hospital)            0.030798            0.092942   

          support  confidence       lift  interest  
3237741  0.030359    0.982270  10.568664  0.889328  
5347777  0.030140    0.982143  10.567301  0.889201  
5349300  0.030031    0.982079  10.566613  0.889137  
1372218  0.031346    0.979452  10.538350  0.886510  
4710745  0.031017    0.979239  10.536055  0.886297  
880248   0.030250    0.978723  10.530510  0.885782  
6550487  0.030031    0.978571  10.528875  0.885630  
3167559  0.031675    0.976351  10.504988  0.883410  
4710999  0.031675    0.976351  10.504988  0.883410  
3167811  0.031346    0.976109  10.502383  0.883168  
3225267  0.031017    0.975862  10.499723  0.882920  
3180159  0.031017    0.975862  10.499723  0.882920  
3168693  0.030907    0.975779  10.498825  0.882837  
7564125  0.030798    0.975694  10.497920  0.882753  
4711760  0.030798    0.975694  10.497920  0.882753  
4693727  0.030579    0.975524  10.496091  0.882583  
6549468  0.030140    0.975177  10.492356  0.882236  
2460693  0.030140    0.975177  10.492356  0.882236  
1372466  0.030140    0.975177  10.492356  0.882236  
1372404  0.030031    0.975089  10.491405  0.882147

Range of top 20 interest values: 0.889328 to 0.882147

Observations:
- (Hair Salon, Hospital) is the sole consequents group type for the top 20 interest value, indicating a significant relationship with the groups of antecedents.

## 3. Analysis of City C

In [28]:
cityC_analysis = rules_cityC.iloc[:,0:7]
cityC_analysis['interest'] = cityC_analysis['confidence'] - cityC_analysis['consequent support']
cityC_analysis.to_csv('cityC.csv', index=False)

cityC_analysis.head()

antecedents    consequents  \
3951142  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4696044  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
3911771  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   
2846234  (Elderly Care Home, Bank, Home Appliances, Acc...  (Real Estate)   
4628213  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   

         antecedent support  consequent support   support  confidence  \
3951142            0.093817            0.396186  0.093817    1.000000   
4696044            0.092279            0.396186  0.092279    1.000000   
3911771            0.102122            0.396186  0.101815    0.996988   
2846234            0.100277            0.396186  0.099969    0.996933   
4628213            0.100277            0.396186  0.099969    0.996933   

             lift  interest  
3951142  2.524068  0.603814  
4696044  2.524068  0.603814  
3911771  2.516466  0.600802  
2846234  2.516326  0.600747  
4628213  2.516326  0.600747

In [29]:
# Analyse the association rules based on the support value
cityC_analysis.sort_values('support', ascending = False).head(20)

antecedents          consequents  antecedent support  \
460    (Transit Station)               (Park)            0.571516   
461               (Park)    (Transit Station)            0.482621   
553  (Building Material)    (Transit Station)            0.472778   
552    (Transit Station)  (Building Material)            0.571516   
526        (Real Estate)    (Transit Station)            0.396186   
527    (Transit Station)        (Real Estate)            0.571516   
502  (Building Material)               (Park)            0.472778   
503               (Park)  (Building Material)            0.482621   
528    (Transit Station)    (Home Appliances)            0.571516   
529    (Home Appliances)    (Transit Station)            0.434943   
476        (Real Estate)               (Park)            0.396186   
477               (Park)        (Real Estate)            0.482621   
844    (Home Appliances)  (Building Material)            0.434943   
845  (Building Material)    (Home Appliances)            0.472778   
554     (Heavy Industry)    (Transit Station)            0.426946   
555    (Transit Station)     (Heavy Industry)            0.571516   
550  (Accountant Office)    (Transit Station)            0.385727   
551    (Transit Station)  (Accountant Office)            0.571516   
814        (Real Estate)  (Building Material)            0.396186   
815  (Building Material)        (Real Estate)            0.472778   

     consequent support   support  confidence      lift  interest  
460            0.482621  0.392187    0.686222  1.421865  0.203601  
461            0.571516  0.392187    0.812620  1.421865  0.241103  
553            0.571516  0.368194    0.778790  1.362673  0.207273  
552            0.472778  0.368194    0.644241  1.362673  0.171464  
526            0.571516  0.349431    0.881988  1.543241  0.310471  
527            0.396186  0.349431    0.611410  1.543241  0.215224  
502            0.482621  0.338357    0.715680  1.482903  0.233059  
503            0.472778  0.338357    0.701083  1.482903  0.228306  
528            0.434943  0.337742    0.590958  1.358702  0.156015  
529            0.571516  0.337742    0.776521  1.358702  0.205004  
476            0.482621  0.336204    0.848602  1.758322  0.365982  
477            0.396186  0.336204    0.696622  1.758322  0.300436  
844            0.472778  0.329745    0.758133  1.603572  0.285355  
845            0.434943  0.329745    0.697463  1.603572  0.262519  
554            0.571516  0.324823    0.760807  1.331207  0.189290  
555            0.426946  0.324823    0.568353  1.331207  0.141408  
550            0.571516  0.320209    0.830144  1.452528  0.258627  
551            0.385727  0.320209    0.560280  1.452528  0.174552  
814            0.472778  0.318056    0.802795  1.698039  0.330017  
815            0.396186  0.318056    0.672739  1.698039  0.276553

Range of top 20 support values: 0.392187 to 0.318056

Observations:
- No notable observation

In [30]:
# Analyse the association rules based on the confidence value
cityC_analysis.sort_values('confidence', ascending = False).head(20)

antecedents    consequents  \
3951142  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4696044  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
3911771  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   
4628213  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   
2846234  (Elderly Care Home, Bank, Home Appliances, Acc...  (Real Estate)   
4098970  (Elderly Care Home, Bank, Home Appliances, Tra...  (Real Estate)   
4749593  (Elderly Care Home, Hair Salon, Hospital, Bank...  (Real Estate)   
3285154  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4751124  (Elderly Care Home, Bank, Hospital, Home Appli...  (Real Estate)   
3951397  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4916481  (Elderly Care Home, Hair Salon, Bank, Hospital...  (Real Estate)   
4185838  (Elderly Care Home, Hair Salon, Bank, Home App...  (Real Estate)   
3285916  (Elderly Care Home, Hair Salon, Home Appliance...  (Real Estate)   
4186601  (Elderly Care Home, Bank, Home Appliances, Acc...  (Real Estate)   
4336494  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4919548  (Elderly Care Home, Hospital, Bank, Home Appli...  (Real Estate)   
4696555  (Elderly Care Home, Bank, Home Appliances, Par...  (Real Estate)   
4831194  (Elderly Care Home, Hair Salon, Bank, Home App...  (Real Estate)   
4699613  (Elderly Care Home, Hair Salon, Bank, Home App...  (Real Estate)   
4832215  (Elderly Care Home, Bank, Home Appliances, Tra...  (Real Estate)   

         antecedent support  consequent support   support  confidence  \
3951142            0.093817            0.396186  0.093817    1.000000   
4696044            0.092279            0.396186  0.092279    1.000000   
3911771            0.102122            0.396186  0.101815    0.996988   
4628213            0.100277            0.396186  0.099969    0.996933   
2846234            0.100277            0.396186  0.099969    0.996933   
4098970            0.098739            0.396186  0.098431    0.996885   
4749593            0.097201            0.396186  0.096893    0.996835   
3285154            0.096278            0.396186  0.095970    0.996805   
4751124            0.096278            0.396186  0.095970    0.996805   
3951397            0.095970            0.396186  0.095663    0.996795   
4916481            0.095355            0.396186  0.095048    0.996774   
4185838            0.095355            0.396186  0.095048    0.996774   
3285916            0.094432            0.396186  0.094125    0.996743   
4186601            0.094432            0.396186  0.094125    0.996743   
4336494            0.094432            0.396186  0.094125    0.996743   
4919548            0.094432            0.396186  0.094125    0.996743   
4696555            0.093817            0.396186  0.093510    0.996721   
4831194            0.093817            0.396186  0.093510    0.996721   
4699613            0.093510            0.396186  0.093202    0.996711   
4832215            0.092894            0.396186  0.092587    0.996689   

             lift  interest  
3951142  2.524068  0.603814  
4696044  2.524068  0.603814  
3911771  2.516466  0.600802  
4628213  2.516326  0.600747  
2846234  2.516326  0.600747  
4098970  2.516205  0.600699  
4749593  2.516081  0.600650  
3285154  2.516004  0.600619  
4751124  2.516004  0.600619  
3951397  2.515978  0.600609  
4916481  2.515926  0.600588  
4185838  2.515926  0.600588  
3285916  2.515847  0.600557  
4186601  2.515847  0.600557  
4336494  2.515847  0.600557  
4919548  2.515847  0.600557  
4696555  2.515793  0.600536  
4831194  2.515793  0.600536  
4699613  2.515765  0.600525  
4832215  2.515710  0.600503

Range of top 20 confidence values: 1.000000 to 0.996689

Observations:
- The sole consequent is (Real Estate) for the top 20 confidence values
- The group of items in antecedents are quite similar to one another.

In [31]:
# Analyse the association rules based on the lift value
cityC_analysis.sort_values('lift', ascending = False).head(20)

antecedents  \
4282590  (Building Material, Hair Salon, Interior Shop,...   
4282783  (Hospital, Bank, Accountant Office, Heavy Indu...   
2877828  (Hospital, Bank, Interior Shop, Accountant Off...   
2877897  (Real Estate, Transit Station, Japanese restau...   
2877953  (Real Estate, Japanese restaurant, Convenience...   
2877772  (Hospital, Bank, Interior Shop, Accountant Off...   
4200523  (Hair Salon, Hospital, Bank, Accountant Office...   
4200630  (Elderly Care Home, Transit Station, Japanese ...   
4282595  (Building Material, Hair Salon, Heavy Industry...   
4282778  (Hospital, Bank, Interior Shop, Accountant Off...   
2877386  (Real Estate, Transit Station, Japanese restau...   
2877323          (Hospital, Bank, Interior Shop, Laundry )   
4200729  (Elderly Care Home, Japanese restaurant, Conve...   
4200424  (Hair Salon, Hospital, Bank, Transit Station, ...   
1469710  (Hospital, Bank, Interior Shop, Accountant Off...   
1469779  (Real Estate, Japanese restaurant, Convenience...   
2877777  (Hospital, Bank, Interior Shop, Transit Statio...   
2877948  (Accountant Office, Japanese restaurant, Conve...   
4200115  (Elderly Care Home, Transit Station, Japanese ...   
4200018  (Hair Salon, Hospital, Bank, Real Estate, Laun...   

                                               consequents  \
4282590  (Hospital, Bank, Accountant Office, Heavy Indu...   
4282783  (Building Material, Hair Salon, Interior Shop,...   
2877828  (Real Estate, Transit Station, Japanese restau...   
2877897  (Hospital, Bank, Interior Shop, Accountant Off...   
2877953  (Hospital, Bank, Interior Shop, Accountant Off...   
2877772  (Real Estate, Japanese restaurant, Convenience...   
4200523  (Elderly Care Home, Transit Station, Japanese ...   
4200630  (Hair Salon, Hospital, Bank, Accountant Office...   
4282595  (Hospital, Bank, Interior Shop, Accountant Off...   
4282778  (Building Material, Hair Salon, Heavy Industry...   
2877386          (Hospital, Bank, Interior Shop, Laundry )   
2877323  (Real Estate, Transit Station, Japanese restau...   
4200729  (Hair Salon, Hospital, Bank, Transit Station, ...   
4200424  (Elderly Care Home, Japanese restaurant, Conve...   
1469710  (Real Estate, Japanese restaurant, Convenience...   
1469779  (Hospital, Bank, Interior Shop, Accountant Off...   
2877777  (Accountant Office, Japanese restaurant, Conve...   
2877948  (Hospital, Bank, Interior Shop, Transit Statio...   
4200115  (Hair Salon, Hospital, Bank, Real Estate, Laun...   
4200018  (Elderly Care Home, Transit Station, Japanese ...   

         antecedent support  consequent support   support  confidence  \
4282590            0.121501            0.142725  0.090126    0.741772   
4282783            0.142725            0.121501  0.090126    0.631466   
2877828            0.134420            0.129499  0.090126    0.670481   
2877897            0.129499            0.134420  0.090126    0.695962   
2877953            0.132882            0.131037  0.090126    0.678241   
2877772            0.131037            0.132882  0.090126    0.687793   
4200523            0.155337            0.113196  0.090741    0.584158   
4200630            0.113196            0.155337  0.090741    0.801630   
4282595            0.130114            0.134420  0.090126    0.692671   
4282778            0.134420            0.130114  0.090126    0.670481   
2877386            0.129499            0.135343  0.090126    0.695962   
2877323            0.135343            0.129499  0.090126    0.665909   
4200729            0.115964            0.152261  0.090741    0.782493   
4200424            0.152261            0.115964  0.090741    0.595960   
1469710            0.134420            0.132882  0.091664    0.681922   
1469779            0.132882            0.134420  0.091664    0.689815   
2877777            0.144571            0.121501  0.090126    0.623404   
2877948            0.121501            0.144571  0.090126    0.741772   
4200115            0.113196            0.156260  0.090

Range of top 20 lift values: 5.197201 to 5.130119

Observations:
- No notable observation

In [32]:
# Analyse the association rules based on the interest value
cityC_analysis.sort_values('interest', ascending = False).head(20)

antecedents  \
2887963  (Elderly Care Home, Transit Station, Accountan...   
2896811  (Elderly Care Home, Transit Station, Accountan...   
1501101  (Elderly Care Home, Accountant Office, Japanes...   
2896854  (Elderly Care Home, Accountant Office, Japanes...   
4282458  (Hair Salon, Hospital, Interior Shop, Accounta...   
4200398  (Elderly Care Home, Hair Salon, Accountant Off...   
4267727  (Hair Salon, Hospital, Interior Shop, Accounta...   
2903443  (Elderly Care Home, Hair Salon, Accountant Off...   
4200418  (Elderly Care Home, Hospital, Transit Station,...   
2905257  (Hospital, Accountant Office, Japanese restaur...   
4265712  (Hospital, Interior Shop, Accountant Office, H...   
530205   (Elderly Care Home, Interior Shop, Japanese re...   
2897361  (Elderly Care Home, Accountant Office, Transit...   
4265642  (Hospital, Interior Shop, Accountant Office, T...   
1506493  (Elderly Care Home, Accountant Office, Japanes...   
1467078  (Elderly Care Home, Interior Shop, Japanese re...   
1467054  (Elderly Care Home, Interior Shop, Transit Sta...   
3148520  (Elderly Care Home, Hair Salon, Interior Shop,...   
2887924  (Elderly Care Home, Transit Station, Accountan...   
4272258  (Elderly Care Home, Hair Salon, Interior Shop,...   

                                  consequents  antecedent support  \
2887963         (Hospital, Bank, Real Estate)            0.100892   
2896811                   (Real Estate, Bank)            0.093202   
1501101         (Bank, Hospital, Real Estate)            0.102430   
2896854  (Real Estate, Bank, Transit Station)            0.094432   
4282458                   (Real Estate, Bank)            0.092279   
4200398         (Hospital, Bank, Real Estate)            0.097816   
4267727  (Real Estate, Bank, Transit Station)            0.099046   
2903443         (Hospital, Bank, Real Estate)            0.099354   
4200418       (Real Estate, Hair Salon, Bank)            0.096586   
2905257       (Real Estate, Hair Salon, Bank)            0.096278   
4265712  (Real Estate, Bank, Transit Station)            0.095970   
530205                    (Real Estate, Bank)            0.095048   
2897361       (Real Estate, Hair Salon, Bank)            0.100892   
4265642                   (Real Estate, Bank)            0.094740   
1506493                   (Real Estate, Bank)            0.094432   
1467078  (Real Estate, Bank, Transit Station)            0.095048   
1467054                   (Real Estate, Bank)            0.093817   
3148520                   (Real Estate, Bank)            0.093817   
2887924                      (Hospital, Bank)            0.099662   
4272258                   (Real Estate, Bank)            0.092587   

         consequent support   support  confidence      lift  interest  
2887963            0.200246  0.093817    0.929878  4.643677  0.729632  
2896811            0.249154  0.091049    0.976898  3.920857  0.727744  
1501101            0.200246  0.095048    0.927928  4.633938  0.727682  
2896854            0.236543  0.091049    0.964169  4.076092  0.727627  
4282458            0.249154  0.090126    0.976667  3.919930  0.727513  
4200398            0.200246  0.090741    0.927673  4.632665  0.727427  
4267727            0.236543  0.095355    0.962733  4.070019  0.726190  
2903443            0.200246  0.091972    0.925697  4.622795  0.725451  
4200418            0.214088  0.090741    0.939490  4.388338  0.725402  
2905257            0.214088  0.090434    0.939297  4.387435  0.725209  
4265712            0.236543  0.092279    0.961538  4.064969  0.724996  
530205             0.249154  0.092587    0.974110  3.909669  0.724956  
2897361            0.214088  0.094740    0.939024  4.386161  0.724936  
4265642            0.249154  0.092279    0.974026  3.909331  0.724872  
1506493            0.249154  0.091972    0.973941  3.908992  0.724787  
1467078            0.236543  0.091357    0.961165  4.063391  0.724622  
1467054            0.249154  0.091357    0.973770  3.908306  0.724616 

Range of top 20 interest values: 0.729632 to 0.724268

Observations:
- (Elderly Care Home) appears very frequently in the antecedents group, perhaps suggesting a high population of elderly or their family members.

## 4. Analysis of City D

In [33]:
cityD_analysis = rules_cityD.iloc[:,0:7]
cityD_analysis['interest'] = cityD_analysis['confidence'] - cityD_analysis['consequent support']
cityD_analysis.to_csv('cityD.csv', index=False)

cityD_analysis.head()

antecedents   consequents  \
711730  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
941177  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
941685  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
943463  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
713368  (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   

        antecedent support  consequent support   support  confidence  \
711730            0.032490            0.230251  0.032308    0.994398   
941177            0.030943            0.230251  0.030761    0.994118   
941685            0.031216            0.230251  0.030943    0.991254   
943463            0.030397            0.230251  0.030124    0.991018   
713368            0.032763            0.230251  0.032399    0.988889   

            lift  interest  
711730  4.318752  0.764147  
941177  4.317535  0.763866  
941685  4.305097  0.761002  
943463  4.304073  0.760767  
713368  4.294827  0.758638

In [34]:
# Analyse the association rules based on the support value
cityD_analysis.sort_values('support', ascending = False).head(20)

antecedents          consequents  antecedent support  \
1138     (Heavy Industry)  (Building Material)            0.334274   
1139  (Building Material)     (Heavy Industry)            0.322625   
948     (Home Appliances)  (Building Material)            0.331361   
949   (Building Material)    (Home Appliances)            0.322625   
952      (Heavy Industry)    (Home Appliances)            0.334274   
953     (Home Appliances)     (Heavy Industry)            0.331361   
643     (Transit Station)     (Heavy Industry)            0.308063   
642      (Heavy Industry)    (Transit Station)            0.334274   
638     (Transit Station)  (Building Material)            0.308063   
639   (Building Material)    (Transit Station)            0.322625   
612     (Transit Station)    (Home Appliances)            0.308063   
613     (Home Appliances)    (Transit Station)            0.331361   
943     (Home Appliances)             (Church)            0.331361   
942              (Church)    (Home Appliances)            0.360939   
1073  (Building Material)         (Hair Salon)            0.322625   
1072         (Hair Salon)  (Building Material)            0.230251   
1111  (Building Material)             (Church)            0.322625   
1110             (Church)  (Building Material)            0.360939   
627     (Transit Station)         (Hair Salon)            0.308063   
626          (Hair Salon)    (Transit Station)            0.230251   

      consequent support   support  confidence      lift  interest  
1138            0.322625  0.172370    0.515655  1.598312  0.193030  
1139            0.334274  0.172370    0.534274  1.598312  0.200000  
948             0.322625  0.171460    0.517440  1.603846  0.194816  
949             0.331361  0.171460    0.531453  1.603846  0.200091  
952             0.331361  0.162268    0.485434  1.464969  0.154073  
953             0.334274  0.162268    0.489701  1.464969  0.155427  
643             0.334274  0.155442    0.504579  1.509478  0.170305  
642             0.308063  0.155442    0.465015  1.509478  0.156952  
638             0.322625  0.154350    0.501034  1.552993  0.178409  
639             0.308063  0.154350    0.478420  1.552993  0.170357  
612             0.331361  0.147161    0.477696  1.441615  0.146334  
613             0.308063  0.147161    0.444109  1.441615  0.136045  
943             0.360939  0.145249    0.438341  1.214446  0.077402  
942             0.331361  0.145249    0.402421  1.214446  0.071059  
1073            0.230251  0.143884    0.445980  1.936929  0.215729  
1072            0.322625  0.143884    0.624901  1.936929  0.302277  
1111            0.360939  0.143611    0.445134  1.233266  0.084195  
1110            0.322625  0.143611    0.397882  1.233266  0.075257  
627             0.230251  0.142064    0.461152  2.002822  0.230901  
626             0.308063  0.142064    0.616996  2.002822  0.308933

Range of top 20 support values: 0.172370 to 0.142064

Observations:
- No notable observations

In [35]:
# Analyse the association rules based on the confidence value
cityD_analysis.sort_values('confidence', ascending = False).head(20)

antecedents   consequents  \
711730   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
941177   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
941685   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
943463   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
713368   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
723826   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
712990   (Hospital, Bank, Transit Station, Accountant O...  (Hair Salon)   
967339   (Hospital, Bank, Accountant Office, Heavy Indu...  (Hair Salon)   
723952   (Hospital, Transit Station, Japanese restauran...  (Hair Salon)   
941431   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
712360   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
963529   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
794764   (Hospital, Bank, Transit Station, Accountant O...  (Hair Salon)   
711478   (Hospital, Transit Station, Japanese restauran...  (Hair Salon)   
716770   (Hospital, Accountant Office, Japanese restaur...  (Hair Salon)   
717022   (Hospital, Japanese restaurant, Heavy Industry...  (Hair Salon)   
754822   (Hospital, Accountant Office, Japanese restaur...  (Hair Salon)   
713242   (Hospital, Transit Station, Accountant Office,...  (Hair Salon)   
711982   (Hospital, Transit Station, Japanese restauran...  (Hair Salon)   
1005442  (Hospital, Bank, Transit Station, Accountant O...  (Hair Salon)   

         antecedent support  consequent support   support  confidence  \
711730             0.032490            0.230251  0.032308    0.994398   
941177             0.030943            0.230251  0.030761    0.994118   
941685             0.031216            0.230251  0.030943    0.991254   
943463             0.030397            0.230251  0.030124    0.991018   
713368             0.032763            0.230251  0.032399    0.988889   
723826             0.031580            0.230251  0.031216    0.988473   
712990             0.031398            0.230251  0.031034    0.988406   
967339             0.031216            0.230251  0.030852    0.988338   
723952             0.031034            0.230251  0.030670    0.988270   
941431             0.030397            0.230251  0.030033    0.988024   
712360             0.034765            0.230251  0.034310    0.986911   
963529             0.033582            0.230251  0.033127    0.986450   
794764             0.033491            0.230251  0.033036    0.986413   
711478             0.033400            0.230251  0.032945    0.986376   
716770             0.033218            0.230251  0.032763    0.986301   
717022             0.032308            0.230251  0.031853    0.985915   
754822             0.032308            0.230251  0.031853    0.985915   
713242             0.032035            0.230251  0.031580    0.985795   
711982             0.031853            0.230251  0.031398    0.985714   
1005442            0.031853            0.230251  0.031398    0.985714   

             lift  interest  
711730   4.318752  0.764147  
941177   4.317535  0.763866  
941685   4.305097  0.761002  
943463   4.304073  0.760767  
713368   4.294827  0.758638  
723826   4.293019  0.758221  
712990   4.292728  0.758155  
967339   4.292435  0.758087  
723952   4.292138  0.758019  
941431   4.291070  0.757773  
712360   4.286236  0.756660  
963529   4.284234  0.756199  
794764   4.284074  0.756162  
711478   4.283913  0.756125  
716770   4.283589  0.756050  
717022   4.281913  0.755664  
754822   4.281913  0.755664  
713242   4.281391  0.755544  
711982   4.281039  0.755463  
1005442  4.281039  0.755463

Range of top 20 confidence values: 0.994398 to 0.985714

Observations:
- (Hair Salon) is once again the sole consequent
- The antecedent group is also very similar to one another.

In [36]:
# Analyse the association rules based on the lift value
cityD_analysis.sort_values('lift', ascending = False).head(20)

antecedents  \
943565   (Fishing, Hair Salon, Accountant Office, Trans...   
943600   (Real Estate, Hospital, Laundry , Japanese res...   
1005583  (Hair Salon, Bank, Transit Station, Heavy Indu...   
1005794  (Hospital, Japanese restaurant, Accountant Off...   
1005723  (Hair Salon, Hospital, Laundry , Japanese rest...   
1005654  (Bank, Transit Station, Accountant Office, Hea...   
1005069  (Hair Salon, Bank, Transit Station, Accountant...   
1005288  (Hospital, Laundry , Japanese restaurant, Buil...   
1005580  (Hair Salon, Bank, Transit Station, Accountant...   
1005797  (Real Estate, Hospital, Laundry , Japanese res...   
1005500  (Hair Salon, Bank, Transit Station, Accountant...   
1005877          (Hospital, Laundry , Japanese restaurant)   
1005578  (Hair Salon, Bank, Transit Station, Accountant...   
1005799  (Heavy Industry, Hospital, Laundry , Japanese ...   
705637           (Hospital, Laundry , Japanese restaurant)   
705588   (Bank, Interior Shop, Accountant Office, Trans...   
1005584  (Hair Salon, Bank, Transit Station, Real Estat...   
1005793  (Heavy Industry, Hospital, Japanese restaurant...   
1005106  (Hair Salon, Hospital, Japanese restaurant, La...   
1005251  (Real Estate, Bank, Transit Station, Accountan...   

                                               consequents  \
943565   (Real Estate, Hospital, Laundry , Japanese res...   
943600   (Fishing, Hair Salon, Accountant Office, Trans...   
1005583  (Hospital, Japanese restaurant, Accountant Off...   
1005794  (Hair Salon, Bank, Transit Station, Heavy Indu...   
1005723  (Bank, Transit Station, Accountant Office, Hea...   
1005654  (Hair Salon, Hospital, Laundry , Japanese rest...   
1005069  (Hospital, Laundry , Japanese restaurant, Buil...   
1005288  (Hair Salon, Bank, Transit Station, Accountant...   
1005580  (Real Estate, Hospital, Laundry , Japanese res...   
1005797  (Hair Salon, Bank, Transit Station, Accountant...   
1005500          (Hospital, Laundry , Japanese restaurant)   
1005877  (Hair Salon, Bank, Transit Station, Accountant...   
1005578  (Heavy Industry, Hospital, Laundry , Japanese ...   
1005799  (Hair Salon, Bank, Transit Station, Accountant...   
705637   (Bank, Interior Shop, Accountant Office, Trans...   
705588           (Hospital, Laundry , Japanese restaurant)   
1005584  (Heavy Industry, Hospital, Japanese restaurant...   
1005793  (Hair Salon, Bank, Transit Station, Real Estat...   
1005106  (Real Estate, Bank, Transit Station, Accountan...   
1005251  (Hair Salon, Hospital, Japanese restaurant, La...   

         antecedent support  consequent support   support  confidence  \
943565             0.049691            0.048689  0.030124    0.606227   
943600             0.048689            0.049691  0.030124    0.618692   
1005583            0.051147            0.050055  0.031398    0.613879   
1005794            0.050055            0.051147  0.031398    0.627273   
1005723            0.053695            0.047688  0.031398    0.584746   
1005654            0.047688            0.053695  0.031398    0.658397   
1005069            0.051511            0.048963  0.030761    0.597173   
1005288            0.048963            0.051511  0.030761    0.628253   
1005580            0.052967            0.048689  0.031398    0.592784   
1005797            0.048689            0.052967  0.031398    0.644860   
1005500            0.044594            0.057972  0.031398    0.704082   
1005877            0.057972            0.044594  0.031398    0.541601   
1005578            0.051511            0.050237  0.031398    0.609541   
1005799            0.050237            0.051511  0.031398    0.625000   
705637             0.057972            0.043502  0.030579    0.527473   
705588             0.043502            0.057972  0.030579    0.702929   
1005584            0.053513            0.048416  0.031398    0.586735   
1005793            0.048416            0.053513  0.031398    0.648496   
1005106            0.045868            0.055424  0.030

Range of top 20 lift values: 12.450885 to 12.100060

Observations:
- No notable observations.

In [37]:
# Analyse the association rules based on the interest value
cityD_analysis.sort_values('interest', ascending = False).head(20)

antecedents  \
943485   (Hospital, Transit Station, Accountant Office,...   
773721   (Hair Salon, Bank, Transit Station, Accountant...   
1005471  (Hospital, Bank, Transit Station, Accountant O...   
450478   (Hair Salon, Bank, Transit Station, Real Estat...   
164048            (Real Estate, Grocery Store, Drug Store)   
454074   (Hair Salon, Bank, Accountant Office, Real Est...   
941199   (Hospital, Transit Station, Accountant Office,...   
451780   (Hair Salon, Bank, Transit Station, Accountant...   
773846   (Hair Salon, Bank, Transit Station, Heavy Indu...   
454322   (Hair Salon, Accountant Office, Heavy Industry...   
774099   (Hair Salon, Accountant Office, Transit Statio...   
941707   (Hospital, Transit Station, Accountant Office,...   
450494    (Real Estate, Drug Store, Bank, Transit Station)   
454090   (Real Estate, Drug Store, Bank, Accountant Off...   
164367   (Real Estate, Elderly Care Home, Transit Stati...   
454197   (Hair Salon, Bank, Heavy Industry, Real Estate...   
454136   (Hair Salon, Bank, Real Estate, Drug Store, Bu...   
168027         (Real Estate, Hair Salon, Bank, Drug Store)   
450541   (Bank, Transit Station, Accountant Office, Rea...   
450105   (Elderly Care Home, Hair Salon, Transit Statio...   

                    consequents  antecedent support  consequent support  \
943485   (Hair Salon, Laundry )            0.031580            0.124317   
773721               (Hospital)            0.032126            0.143975   
1005471  (Hair Salon, Laundry )            0.033127            0.124317   
450478               (Hospital)            0.032854            0.143975   
164048   (Hair Salon, Hospital)            0.033127            0.105934   
454074               (Hospital)            0.032581            0.143975   
941199   (Hair Salon, Laundry )            0.032490            0.124317   
451780               (Hospital)            0.034037            0.143975   
773846               (Hospital)            0.031125            0.143975   
454322               (Hospital)            0.033309            0.143975   
774099               (Hospital)            0.033036            0.143975   
941707   (Hair Salon, Laundry )            0.032763            0.124317   
450494   (Hair Salon, Hospital)            0.034310            0.105934   
454090   (Hair Salon, Hospital)            0.034037            0.105934   
164367               (Hospital)            0.031762            0.143975   
454197               (Hospital)            0.031671            0.143975   
454136               (Hospital)            0.031580            0.143975   
168027               (Hospital)            0.036039            0.143975   
450541               (Hospital)            0.031216            0.143975   
450105               (Hospital)            0.031216            0.143975   

          support  confidence      lift  interest  
943485   0.030124    0.953890  7.673022  0.829573  
773721   0.031125    0.968839  6.729202  0.824863  
1005471  0.031398    0.947802  7.624049  0.823485  
450478   0.031762    0.966759  6.714758  0.822784  
164048   0.030761    0.928571  8.765587  0.822638  
454074   0.031489    0.966480  6.712824  0.822505  
941199   0.030761    0.946779  7.615816  0.822461  
451780   0.032854    0.965241  6.704212  0.821265  
773846   0.030033    0.964912  6.701932  0.820937  
454322   0.032126    0.964481  6.698935  0.820506  
774099   0.031853    0.964187  6.696897  0.820212  
941707   0.030943    0.944444  7.597039  0.820127  
450494   0.031762    0.925729  8.738759  0.819796  
454090   0.031489    0.925134  8.733135  0.819200  
164367   0.030579    0.962751  6.686918  0.818775  
454197   0.030488    0.962644  6.686175  0.818668  
454136   0.030397    0.962536  6.685427  0.818561  
168027   0.034674    0.962121  6.682546  0.818146  
450541   0.030033    0.962099  6.682393  0.818124  
450105   0.030033    0.962099  6.682393  0.818124

Range of top 20 interest values: 0.829573 to 0.818124

Observations:
- (Hair Salon), (Laundry) and (Hospital) are the sole consequents